src/ui/SmartSnippet/SmartSnippetCollapsibleSuggestion.ts

```diff
@@ -0,0 +1,112 @@
+import { IPartialQuestionAnswerResponse } from '../../rest/QuestionAnswerResponse';
+import { uniqueId } from 'underscore';
+import { AccessibleButton } from '../../utils/AccessibleButton';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { attachShadow } from '../../misc/AttachShadowPolyfill';
+import { $$, Dom } from '../../utils/Dom';
+import { l } from '../../strings/Strings';
+
+const QUESTION_CLASSNAME = `coveo-smart-snippet-suggestions-question`;
+const QUESTION_TITLE_CLASSNAME = `${QUESTION_CLASSNAME}-title`;
+const QUESTION_TITLE_LABEL_CLASSNAME = `${QUESTION_TITLE_CLASSNAME}-label`;
+const QUESTION_TITLE_CHECKBOX_CLASSNAME = `${QUESTION_TITLE_CLASSNAME}-checkbox`;
+const QUESTION_SNIPPET_CLASSNAME = `${QUESTION_CLASSNAME}-snippet`;
+const QUESTION_SNIPPET_HIDDEN_CLASSNAME = `${QUESTION_SNIPPET_CLASSNAME}-hidden`;
+const SHADOW_CLASSNAME = `${QUESTION_SNIPPET_CLASSNAME}-content`;
+
+export class SmartSnippetCollapsibleSuggestion {
+  private readonly labelId = uniqueId(QUESTION_TITLE_LABEL_CLASSNAME);
+  private readonly snippetId = uniqueId(QUESTION_SNIPPET_CLASSNAME);
+  private readonly checkboxId = uniqueId(QUESTION_TITLE_CHECKBOX_CLASSNAME);
+  private contentLoaded: Promise<void>;
```

Comment:
Should this `Promise` return a `boolean` instead of `void`?

Reply:
When it resolves, the content is always loaded. The value would always be `true`.